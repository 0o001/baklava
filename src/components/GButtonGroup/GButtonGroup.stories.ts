import { storiesOf } from '@storybook/vue';
import GButtonGroup from './GButtonGroup.vue';
import GButton from '../GButton/GButton.vue';
import { action } from '@storybook/addon-actions';

storiesOf('GButtonGroup', module)
  .addParameters({ component: GButtonGroup })
  .add('small', () => ({
    components: { GButtonGroup, GButton },
    data () {
      return {
        buttons: [1, 2, 3, 4, 5],
        activeButton: 3,
      };
    },
    methods: {
      buttonClick (page: number): void {
        const self: any = this;
        self.activeButton = page;
        action('Button Click')(page);
      },
    },
    template: `
      <GButtonGroup>
        <GButton
          v-for="button in buttons"
          size="small"
          :variant="activeButton === button ? 'secondary' : 'light'"
          @click="buttonClick(button)"
        >
          {{button}}
        </GButton>
      </GButtonGroup>
    `,
  }))
  .add('medium', () => ({
    components: { GButtonGroup, GButton },
    data () {
      return {
        buttons: [1, 2, 3, 4, 5],
        activeButton: 3,
      };
    },
    methods: {
      buttonClick (page: number): void {
        const self: any = this;
        self.activeButton = page;
        action('Button Click')(page);
      },
    },
    template: `
      <GButtonGroup>
        <GButton
          v-for="button in buttons"
          size="medium"
          :variant="activeButton === button ? 'secondary' : 'light'"
          @click="buttonClick(button)"
        >
          {{button}}
        </GButton>
      </GButtonGroup>
    `,
  }))
  .add('big', () => ({
    components: { GButtonGroup, GButton },
    data () {
      return {
        buttons: [1, 2, 3, 4, 5],
        activeButton: 3,
      };
    },
    methods: {
      buttonClick (page: number): void {
        const self: any = this;
        self.activeButton = page;
        action('Button Click')(page);
      },
    },
    template: `
      <GButtonGroup>
        <GButton
          v-for="button in buttons"
          size="big"
          :variant="activeButton === button ? 'secondary' : 'light'"
          @click="buttonClick(button)"
        >
          {{button}}
        </GButton>
      </GButtonGroup>
    `,
  }))

;
