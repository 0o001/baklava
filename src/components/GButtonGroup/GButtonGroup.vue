<template>
  <div :class="classNames">
    <slot />
  </div>
</template>

<script lang="ts">

import $_ from '../../utils/dashUtil.ts';

export default {
  name: 'GButtonGroup',

  props: {
    variant: {
      type: String,
      default: 'light',
    },
  },
  computed: {
    classNames () {
      return {
        'g-button-group': true,
        [$_(this.variant)]: true,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
.g-button-group {
  display: flex;
  border-radius: var(--radius-md);
  width: max-content;

  &.-primary {
    background-color: var(--orange-500);
  }

  &.-primary-darker {
    background-color: var(--orange-300);
  }

  &.-secondary {
    background-color: var(--main-grey-500);
  }

  &.-secondary-darker {
    background-color: var(--main-grey-300);
  }

  &.-tertiary {
    background-color: var(--light-grey-500);
  }

  &.-tertiary-darker {
    background-color: var(--light-grey-300);
  }

  &.-success {
    background-color: var(--green-500);
  }

  &.-success-darker {
    background-color: var(--green-300);
  }

  &.-info {
    background-color: var(--blue-500);
  }

  &.-info-darker {
    background-color: var(--blue-300);
  }

  &.-warning {
    background-color: var(--orange-500);
  }

  &.-warning-darker {
    background-color: var(--orange-300);
  }

  &.-danger {
    background-color: var(--red-500);
  }

  &.-danger-darker {
    background-color: var(--red-300);
  }

  &.-light {
    background-color: var(--bg-grey-500);
  }

  &.-light-darker {
    background-color: var(--bg-grey-300);
  }

  &.-dark {
    background-color: var(--main-grey-500);
  }

  &.-dark-darker {
    background-color: var(--main-grey-300);
  }

  &.-gradient-purple {
    background-image: linear-gradient(180deg,white,white),linear-gradient(100deg,var(--purple-500),var(--blue-500));
    background-origin: border-box;
    background-clip: content-box,border-box;
    text-decoration: none !important;
  }

}
</style>
