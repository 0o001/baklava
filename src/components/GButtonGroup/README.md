# GButtonGroup

## Slots

| Name      | Description |
| --------- | ----------- |
| `default` | &nbsp;      |

## Props

| Name      | Type     | Description | Default   |
| --------- | -------- | ----------- | --------- |
| `variant` | `String` |             | `"light"` |

## Computed Properties

| Name         | Type     | Description                 |
| ------------ | -------- | --------------------------- |
| `classNames` | `object` | **Dependencies:** `variant` |

