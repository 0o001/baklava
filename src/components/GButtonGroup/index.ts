import GButtonGroup from './GButtonGroup.vue';

export default GButtonGroup;
