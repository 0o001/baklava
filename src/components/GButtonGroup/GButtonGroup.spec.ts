import { mount } from '@vue/test-utils';
import GButtonGroup from './';

describe('GButtonGroup', () => {
  let wrapper: any;

  const renderWrapper = (options: Object = {}) => {
    wrapper = mount(GButtonGroup, {
      ...options,
    });
  };

  it('should match snapshot', () => {
    renderWrapper();
    expect(wrapper.find('div').classes())
      .toEqual(['g-button-group', '-light']);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with correct classess', () => {
    renderWrapper({
      propsData: { variant: 'primary' },
    });
    expect(wrapper.find('div').classes())
      .toEqual(['g-button-group', '-primary']);
  });
});
