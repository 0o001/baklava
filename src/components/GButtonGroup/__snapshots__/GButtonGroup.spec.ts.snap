// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GButtonGroup should match snapshot 1`] = `
<div
  class="g-button-group -light"
/>
`;
